app/models/Property.test.ts

```diff
@@ -1,56 +1,183 @@
+  });
+  
+  it("correctly calculates the landPrice", () => {
+    expect(property.landPrice).toBeCloseTo(32575);
+  });
+  
+  it("correctly calculates the landToTotalRatio", () => {
+    expect(property.landToTotalRatio).toBeCloseTo(0.1446);
+  });
+  
+  it("correctly calculates the values even for number of bedroooms exceeding the max ", () => {
+    property = new Property({
+      postcode: "WV8 1HG",
+      houseType: "T",
+      numberOfBedrooms: 20,
+      age: 11,
+      size: 88,
+      maintenancePercentage: 0.02,
+      newBuildPricePerMetre: 2120,
+      averageMarketPrice: 218091.58,
+      itl3: "TLG24",
+    });
+    expect(property).toBeInstanceOf(Property);
+  });
+
+  it("correctly returns newBuildPrice if newbuild", () => {
+    property = new Property({
+      postcode: "WV8 1HG",
+      houseType: "T",
+      numberOfBedrooms: 20,
+      age: 1,
+      size: 88,
+      maintenancePercentage: 0.02,
+      newBuildPricePerMetre: 2120,
+      averageMarketPrice: 218091.58,
+      itl3: "TLG24",
+    });
+
+    expect(property.depreciatedBuildPrice).toBe(property.newBuildPrice);
+  });
+
+  describe('depreciation calculations (existing build)', () => {
+
+    it("correctly calculates newComponentValue for foundations", () => {
+      const result = property.calculateComponentValue(
+        'foundations',
+        property.newBuildPrice,
+        property.age,
+        MAINTENANCE_LEVELS[0]
+      );
+
+      const expectedNewComponentValue = 
+        property.newBuildPrice * 
+        HOUSE_BREAKDOWN_PERCENTAGES.foundations.percentageOfHouse;
+
+      expect(result.newComponentValue).toBe(expectedNewComponentValue);
+      expect(result.maintenanceAddition).toBe(0); // Foundations should have no maintenance
+    });
+
+    it("correctly calculates depreciationFactor for internal linings", () => {
+      const result = property.calculateComponentValue(
+        'internalLinings',
+        property.newBuildPrice,
+        property.age,
+        MAINTENANCE_LEVELS[0]
+      );
+
+      const expectedDepreciationFactor = 
+        1 - (HOUSE_BREAKDOWN_PERCENTAGES.internalLinings.depreciationPercentageYearly * property.age);
+
+      expect(result.depreciationFactor).toBe(expectedDepreciationFactor);
+    });
+
+    it("correctly calculates maintenanceAddition for electrical appliances", () => {
+      const result = property.calculateComponentValue(
+        'electricalAppliances',
+        property.newBuildPrice,
+        property.age,
+        MAINTENANCE_LEVELS[0]
+      );
+
+      const expectedMaintenanceAddition = 
+        MAINTENANCE_LEVELS[0] * 
+        property.newBuildPrice * 
+        property.age * 
+        HOUSE_BREAKDOWN_PERCENTAGES.electricalAppliances.percentOfMaintenanceYearly;
+
+      expect(result.maintenanceAddition).toBe(expectedMaintenanceAddition);
+    });
+
+    it("correctly calculates depreciatedComponentValue for ventilation services", () => {
```

Comment:
See above comment re:duplicating logic - at the level of complexity it feels very tricky to have duplicated logic here and in the class.